pkg/cloudprovider/aws/launchtemplate_test.go

```diff
@@ -1208,6 +1218,135 @@ var _ = Describe("LaunchTemplates", func() {
+					},
+				}})
+				nodeTemplate := test.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{
+					UserData:    nil,
+					AMISelector: map[string]string{"karpenter.sh/discovery": "my-cluster"},
+					AWS:         *provider,
+				})
+				ExpectApplied(ctx, env.Client, nodeTemplate)
+				newProvisioner := test.Provisioner(test.ProvisionerOptions{
+					ProviderRef: &v1alpha5.ProviderRef{Name: nodeTemplate.Name},
+					Requirements: []v1.NodeSelectorRequirement{
+						{
+							Key:      v1.LabelArchStable,
+							Operator: v1.NodeSelectorOpIn,
+							Values:   []string{v1alpha5.ArchitectureAmd64},
+						},
+					},
+				})
+				ExpectApplied(ctx, env.Client, newProvisioner)
+				pod := ExpectProvisioned(ctx, env.Client, controller, test.UnschedulablePod())[0]
+				ExpectScheduled(ctx, env.Client, pod)
+				Expect(fakeEC2API.CalledWithCreateLaunchTemplateInput.Len()).To(Equal(1))
+				input := fakeEC2API.CalledWithCreateLaunchTemplateInput.Pop()
+				Expect("ami-456").To(Equal(*input.LaunchTemplateData.ImageId))
+			})
+			It("should create a launch template with the newest compatible AMI when multiple amis are discovered via the Name tag", func() {
+				opts.AWSENILimitedPodDensity = false
+				fakeEC2API.DescribeImagesOutput.Set(&ec2.DescribeImagesOutput{Images: []*ec2.Image{
+					{
+						ImageId:      aws.String("ami-123"),
+						Architecture: aws.String("x86_64"),
+						CreationDate: aws.String("2020-01-01T12:00:00Z"),
+					},
+					{
+						ImageId:      aws.String("ami-456"),
+						Architecture: aws.String("x86_64"),
+						CreationDate: aws.String("2021-01-01T12:00:00Z"),
+					},
+					{
+						// Incompatible because required ARM64
+						ImageId:      aws.String("ami-789"),
+						Architecture: aws.String("arm64"),
+						CreationDate: aws.String("2022-01-01T12:00:00Z"),
+					},
+				}})
+				nodeTemplate := test.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{
+					UserData:    nil,
+					AMISelector: map[string]string{"Name": "ami-name"},
+					AWS:         *provider,
+				})
+				ExpectApplied(ctx, env.Client, nodeTemplate)
+				newProvisioner := test.Provisioner(test.ProvisionerOptions{
+					ProviderRef: &v1alpha5.ProviderRef{Name: nodeTemplate.Name},
+					Requirements: []v1.NodeSelectorRequirement{
+						{
+							Key:      v1.LabelArchStable,
+							Operator: v1.NodeSelectorOpIn,
+							Values:   []string{v1alpha5.ArchitectureAmd64},
+						},
+					},
+				})
+				ExpectApplied(ctx, env.Client, newProvisioner)
+				pod := ExpectProvisioned(ctx, env.Client, controller, test.UnschedulablePod())[0]
+				ExpectScheduled(ctx, env.Client, pod)
+				Expect(fakeEC2API.CalledWithCreateLaunchTemplateInput.Len()).To(Equal(1))
+				input := fakeEC2API.CalledWithCreateLaunchTemplateInput.Pop()
+				Expect("ami-456").To(Equal(*input.LaunchTemplateData.ImageId))
+			})
+			It("should create a launch template with the newest compatible AMI when multiple amis are discovered via the aws-ids tag", func() {
```

Comment:
Same comment here, because the response is mocked, this has no difference to the tests above it